apps/ledger-live-mobile/src/screens/BleDevicePairingFlow/index.tsx

```diff
@@ -134,46 +133,8 @@ export const BleDevicePairingFlow = ({ navigation, route }: Props) => {
     [navigateInput, navigation, navigationType, pathToDeviceParam],
   );
 
-  const handleGoBackFromScanning = useCallback(() => {
-    const routes = navigation.getState().routes;
-
-    const isNavigationFromDeeplink =
-      routes[routes.length - 1]?.params === undefined;
-
-    if (!isNavigationFromDeeplink) {
-      navigation.goBack();
-    } else {
-      navigation.reset({
-        index: 0,
-        routes: [
-          {
-            // @ts-expect-error is fixed in a future screen
-            name: NavigatorName.BaseOnboarding,
-            state: {
-              routes: [
-                {
-                  name: ScreenName.OnboardingWelcome,
-                },
-              ],
-            },
-          },
-        ],
-      });
-    }
-  }, [navigation]);
```

Comment:
Not needed anymore ?
I would expect us to do something similar if we go to this screen with a deeplink

Reply:
It was a hacky workaround of the fact that the `initialRouteName` was badly set for the onboarding deeplinking config (see that change in the `DeeplinksProvider`).
Having that `initialRouteName` set correctly ensures that we don't land on an incorrect page if we deeplink to something in a non-onboarding LLM and do a `navigation.goBack()`.

Reply:
Nice

Reply:
1. Non-onboarded LLM, deeplink to that screen, cold start: previous screen will be the app's welcome screen
2. Non-onboarded LLM, deeplink to that screen, warm start (app already running): previous screen will be whatever screen was displayed before opening the deeplink.
3. Onboarded LLM, deeplink to that screen, cold start: previous screen will be the Wallet screen.
4. Onboarded LLM, deeplink to that screen, cold start: previous screen will be whatever screen was displayed before opening the deeplink.